src/frontend/src/common/components/library/data_table/index.tsx

```diff
@@ -61,14 +61,15 @@ function sortData(
 ): TableItem[] {
   return data.sort((a, b): number => {
     let order = 0;
+    // Keep failed samples at the bottom of the table
```

Comment:
I'm a bit confused by this comment. Could you expand on it a bit? Is it the case that if a sample was failed, its upload date in the table is `null` and those are the only samples with `null` upload dates, so we can use null-ness in date as a proxy for being a failed sample? I think that's gotta be what's going on, but it's a bit of a non-intuitive jump to understand. If I'm right, could you add another sentence that explains as much?

If I'm wrong, uh, what is going on then?